listenbrainz/webserver/static/js/src/user/Listens.tsx

```diff
@@ -243,9 +248,17 @@ export default class Listens extends React.Component<
     }
   };
 
-  receiveNewPlayingNow = (newPlayingNow: string): void => {
-    const playingNow = JSON.parse(newPlayingNow) as Listen;
-    playingNow.playing_now = true;
+  receiveNewPlayingNow = async (newPlayingNow: Listen): Promise<void> => {
+    const playingNow = newPlayingNow;
+    const { APIService } = this.context;
+    const metadata = await APIService.lookupRecordingMetadata(
+      playingNow.track_metadata.track_name,
+      playingNow.track_metadata.artist_name,
+      false
+    );
+    playingNow.track_metadata.mbid_mapping = metadata as MbidMapping;
+
+    this.loadFeedbackForNowPlaying(playingNow);
```

Comment:
This should be wrapped in a try-catch block in case the request fails.

Reply:
Done.